napari/_qt/dialogs/qt_reader_dialog.py

```diff
@@ -67,9 +67,25 @@ def setup_ui(self, error_message, readers):
 
         # checkbox to remember the choice (doesn't pop up for folders with no extension)
         if self._extension:
-            self.persist_checkbox = QCheckBox(
-                f'Remember this choice for files with a {self._extension} extension'
-            )
+            if (
+                '*' + self._extension
+                in get_settings().plugins.extension2reader.keys()
+            ):
+                pref = get_settings().plugins.extension2reader[
```

Comment:
the settings file stores plugin name not display name, so we should get the display name of the plugin here.